include/kframework/lemmas/bitwise-simplification.k

```diff
@@ -20,85 +22,108 @@ module BITWISE-SIMPLIFICATION [symbolic]
-    rule N &Int maxUInt128 => N  requires #rangeUInt(128, N) [simplification]
-    rule maxUInt128 &Int N => N  requires #rangeUInt(128, N) [simplification]
-    rule N &Int maxUInt120 => N  requires #rangeUInt(120, N) [simplification]
-    rule maxUInt120 &Int N => N  requires #rangeUInt(120, N) [simplification]
-    rule N &Int maxUInt112 => N  requires #rangeUInt(112, N) [simplification]
-    rule maxUInt112 &Int N => N  requires #rangeUInt(112, N) [simplification]
-    rule N &Int maxUInt104 => N  requires #rangeUInt(104, N) [simplification]
-    rule maxUInt104 &Int N => N  requires #rangeUInt(104, N) [simplification]
-    rule N &Int maxUInt96  => N  requires #rangeUInt(96,  N) [simplification]
-    rule maxUInt96 &Int N  => N  requires #rangeUInt(96,  N) [simplification]
-    rule N &Int maxUInt88  => N  requires #rangeUInt(88,  N) [simplification]
-    rule maxUInt88 &Int N  => N  requires #rangeUInt(88,  N) [simplification]
-    rule N &Int maxUInt80  => N  requires #rangeUInt(80,  N) [simplification]
-    rule maxUInt80 &Int N  => N  requires #rangeUInt(80,  N) [simplification]
-    rule N &Int maxUInt72  => N  requires #rangeUInt(72,  N) [simplification]
-    rule maxUInt72 &Int N  => N  requires #rangeUInt(72,  N) [simplification]
-    rule N &Int maxUInt64  => N  requires #rangeUInt(64,  N) [simplification]
-    rule maxUInt64 &Int N  => N  requires #rangeUInt(64,  N) [simplification]
-    rule N &Int maxUInt56  => N  requires #rangeUInt(56,  N) [simplification]
-    rule maxUInt56 &Int N  => N  requires #rangeUInt(56,  N) [simplification]
-    rule N &Int maxUInt48  => N  requires #rangeUInt(48,  N) [simplification]
-    rule maxUInt48 &Int N  => N  requires #rangeUInt(48,  N) [simplification]
-    rule N &Int maxUInt40  => N  requires #rangeUInt(40,  N) [simplification]
-    rule maxUInt40 &Int N  => N  requires #rangeUInt(40,  N) [simplification]
-    rule N &Int maxUInt32  => N  requires #rangeUInt(32,  N) [simplification]
-    rule maxUInt32 &Int N  => N  requires #rangeUInt(32,  N) [simplification]
-    rule N &Int maxUInt24  => N  requires #rangeUInt(24,  N) [simplification]
-    rule maxUInt24 &Int N  => N  requires #rangeUInt(24,  N) [simplification]
-    rule N &Int maxUInt16  => N  requires #rangeUInt(16,  N) [simplification]
-    rule maxUInt16 &Int N  => N  requires #rangeUInt(16,  N) [simplification]
-    rule N &Int maxUInt8   => N  requires #rangeUInt(8,   N) [simplification]
-    rule maxUInt8 &Int N   => N  requires #rangeUInt(8,   N) [simplification]
+    // PETAR: Could the 64 subsequent simplifications be conflated into one?
+    //        Or two, if commutativity can't be made to work? Maybe:
+    // rule [bitwise-and-identity]:
+    //   X &Int Y => Y
+    //   requires 0 <=Int X
+    //    andBool X +Int 1 ==Int 2 ^Int log2Int(X +Int 1)
+    //    andBool #rangeUInt(log2Int(X +Int 1), Y)
+    //   [concrete(X), simplification, comm]
```

Comment:
Did we try this? If it works it would be quite nice to remove all the other ones. I think it should work. `&Int` has the `comm` attribute, so it may work: 
![image](https://user-images.githubusercontent.com/12721972/230655414-18ccb6fe-a942-4d05-8869-e5f930aefab0.png)

Reply:
I don't know how well `concrete(X)` will interact with `comm` though. It may be worth inspecting the generated `definition.kore` to make sure it's behaving as expected.

Reply:
Added an issue, so it can be addressed in a later PR: https://github.com/runtimeverification/evm-semantics/issues/1709